examples/sql_bank/main.go

```diff
@@ -58,21 +60,28 @@ UPDATE bank.accounts
   SET balance = CASE id WHEN $1 THEN balance-$3 WHEN $2 THEN balance+$3 END
   WHERE id IN ($1, $2) AND (SELECT balance >= $3 FROM bank.accounts WHERE id = $1)
 `
-			if _, err := db.Exec(update, from, to, amount); err != nil {
+			start := time.Now()
+			result, err := db.Exec(update, from, to, amount)
+			if err != nil {
 				if log.V(1) {
 					log.Warning(err)
 				}
 				continue
 			}
-			// TODO(pmattis): We should only be updating numTransfers when rows were
-			// actually updated. See #2377.
-			atomic.AddUint64(&numTransfers, 1)
+			if result == driver.ResultNoRows {
```

Comment:
I think fixing the TODO is good, but the right way to do it is by calling result.RowsAffected